### ISSUE TITLE
Async/Await

### ISSUE THREAD

**Issue by OptimusLime** [godot-rust/gdext#640]
As I understand it, the base Godot C++ GDExtension library doesn't handle async/yield. I recognize that gdext doesn't have the functionality either, even though gdnative version of godot-rust was able to do it. 

1. Is it even possible to get an await call to work? 
2. What would be the simplest way to enable this, even if it's not ideal?

Any hints for how to do this in gdext would be amazing, thank you!

**Comment by coder137** [comment]
@OptimusLime Maybe this thread can help? https://discord.com/channels/723850269347283004/1213264585725837364/1213264585725837364

Basically `smol::LocalExecutor` and its `try_tick` function should work here

**Comment by Bromeon** [comment]
There are a few people on Discord who have experimented with async, and there's also prior art in the gdnative project with its `gdnative::task` module.

But for me personally, async support is not a priority in the library, there are many other things I'd like to do first. For example, coming up with a better threading model might also help for some questions in async. So, unless someone is willing to spend the time on this, don't expect there to be much progress this year 🙂

**Comment by Bromeon** [comment]
User @Houtamelo has created an extension crate `gdext_coroutines` which supports not only `async` workflows, but also Rust based coroutines! :chart_with_upwards_trend: 

The project uses nightly Rust and is hosted [here](https://github.com/Houtamelo/gdext_coroutines).

**Comment by TitanNano** [comment]
I think this issue has also been covered by #1043.

**Comment by Bromeon** [comment]
Yep, between your async signal contributions and Houtamelo's [`gdext_coroutines`](https://github.com/Houtamelo/gdext_coroutines) crate, there are now many options to use async in godot-rust. Very cool to see how far we've come, and huge thanks to both of you! 👍
